### ISSUE TITLE
Horizontal Scroll glitch in Pill Tab

### ISSUE THREAD

**Issue by AdamMeza-Bentley** [iTwin/iTwinUI#1870]
### Describe the bug (current behavior)

When the third pill tab is clicked and becomes active, it causes a glitching error where a horizontal scroll bar appears and disappears repeatedly. Video of the behavior is below. Confirmed it's happening on [our live site](https://itwinui.bentley.com/docs/tabs#pill) as well. Bug does not happen to the other two tabs. 
 
https://github.com/iTwin/iTwinUI/assets/148785417/41e081f1-5275-44e1-9da4-9d7fa28b2029

https://github.com/iTwin/iTwinUI/assets/148785417/5c849656-8b74-4f66-ad0a-95a23c921ee5

### Expected Behavior

Expected it to not have any horizontal scroll. 

### Steps To Reproduce

Click on the third tab. Happens every time for me.

**Comment by mayank99** [comment]
I cannot reproduce this. Maybe specific to Windows?

@Ben-Pusey-Bentley Can you confirm?

**Comment by rohan-kadkol** [comment]
> I cannot reproduce this. Maybe specific to Windows?

I could reproduce this on Mac 13.6.1 + Edge 121. But this only happened after I increased the screen width beyond some threshold. So I was able to reproduce it when my browser window was fully maximized.

**Comment by Ben-Pusey-Bentley** [comment]
> I cannot reproduce this. Maybe specific to Windows?
> 
> @Ben-Pusey-Bentley Can you confirm?

https://github.com/iTwin/iTwinUI/assets/141063311/6de029b2-068d-4c3f-874a-1046c9e84b9a

Could not reproduce Adam's issue, but I am getting a separate issue with the page where clicking off the third tabs causes the tabs to become much larger. This only seems to happen when clicking off of the third tab specifically.

**Comment by mayank99** [comment]
@rohan-kadkol Do you want to look into this issue, since you're able to reproduce it?

Since no user has reported it and it's not always reproducible, we can consider it to be medium priority (rather than high priority, like we usually do for user-reported bugs).

**Comment by mayank99** [comment]
> I am getting a separate issue with the page where clicking off the third tabs causes the tabs to become much larger. This only seems to happen when clicking off of the third tab specifically.

This should have been fixed by #1410. Maybe there is some edge case we aren't considering.

**Comment by Ben-Pusey-Bentley** [comment]
> > I am getting a separate issue with the page where clicking off the third tabs causes the tabs to become much larger. This only seems to happen when clicking off of the third tab specifically.
> 
> This should have been fixed by #1410. Maybe there is some edge case we aren't considering.

https://github.com/iTwin/iTwinUI/assets/141063311/e0cc8480-c9de-4c2a-968b-c5221a94fcf8


The tabs seem to be allocating space for a scroll bar, because one does appear for a fraction of a second, but then the scroll bar disappears and the tabs stay larger than they should be. Weird behavior that I found, highlighting the tabs with Chrome dev tools causes them to return to their correct size. Shown in the attached recording.

**Comment by mayank99** [comment]
Weirdly, after #1873 merged, I *can* reproduce this, so I'm reopening this. Can others verify?

(I tried recording my screen, but it happens so fast that it's not noticeable.)

**Comment by Ben-Pusey-Bentley** [comment]
> Weirdly, after #1873 merged, I _can_ reproduce this, so I'm reopening this. Can others verify? I tried recording my screen as proof, but it happens so fast that it's not noticeable.
> 
> I _did_ find a different way to reproduce it though:
> 
>  Screen.Recording.2024-02-26.at.11.32.27.AM.mov

I am able to recreate the clip that you posted, but I am still not able to reproduce the original issue with the scrollbar appearing and disappearing rapidly just by selecting the 3rd tab.
